api/config/v1alpha1/envoygateway_types.go

```diff
@@ -313,6 +320,35 @@ type ExtensionTLS struct {
 	CertificateRef gwapiv1b1.SecretObjectReference `json:"certificateRef"`
 }
 
+// EnvoyGatewayObservability defines the Envoy Gateway Observability configuration.
+type EnvoyGatewayObservability struct {
+
+	// Debug defines Debug Server configuration of Envoy Gateway
+	//
+	// +optional
+	Debug *DebugConfig `json:"debug,omitempty"`
+}
+
+// DebugConfig defines the Envoy Gateway Debug Server configuration.
+type DebugConfig struct {
```

Comment:
enable pprof in production env is dangerous
maybe EG should support enable debug but disabled pprof

Reply:
Agree

Reply:
agree this should be opt in as we start